delta/plugins/composite-views/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/compositeviews/BlazegraphQuery.scala

```diff
@@ -0,0 +1,127 @@
+package ch.epfl.bluebrain.nexus.delta.plugins.compositeviews
+
+import ch.epfl.bluebrain.nexus.delta.plugins.blazegraph.BlazegraphViews
+import ch.epfl.bluebrain.nexus.delta.plugins.blazegraph.client.{BlazegraphClient, SparqlQuery, SparqlResults}
+import ch.epfl.bluebrain.nexus.delta.plugins.compositeviews.model.CompositeViewProjection.SparqlProjection
+import ch.epfl.bluebrain.nexus.delta.plugins.compositeviews.model.CompositeViewRejection.{AuthorizationFailed, ProjectionNotFound, WrappedBlazegraphClientError}
+import ch.epfl.bluebrain.nexus.delta.plugins.compositeviews.model.ProjectionType.SparqlProjectionType
+import ch.epfl.bluebrain.nexus.delta.plugins.compositeviews.model.{CompositeView, CompositeViewRejection}
+import ch.epfl.bluebrain.nexus.delta.sdk.Acls
+import ch.epfl.bluebrain.nexus.delta.sdk.model.IdSegment
+import ch.epfl.bluebrain.nexus.delta.sdk.model.identities.Caller
+import ch.epfl.bluebrain.nexus.delta.sdk.model.projects.ProjectRef
+import ch.epfl.bluebrain.nexus.delta.sourcing.config.ExternalIndexingConfig
+import monix.bio.IO
+
+trait BlazegraphQuery {
+
+  /**
+    * Queries the blazegraph common namespace of the passed composite view.
+    * We check for the caller to have the necessary query permissions on all the views' projections before performing the query.
+    *
+    * @param id         the id of the composite view either in Iri or aliased form
+    * @param project    the project where the view exists
+    * @param query      the sparql query to run
+    */
+  def query(
+      id: IdSegment,
+      project: ProjectRef,
+      query: SparqlQuery
+  )(implicit caller: Caller): IO[CompositeViewRejection, SparqlResults]
```

Comment:
I don't remember what we have done for other plugins but `CompositeViewRejection` is a little large.
But it is a detail and it could be addressed later

Reply:
Here you have, some being as long or longer than `CompositeViewRejection`:
- ElasticSearchViewRejection
- BlazegraphViewRejection
- ArchiveRejection